pkg/cmd/dedicated/listclusters/list.go

```diff
@@ -0,0 +1,137 @@
+}
+
+type options struct {
+	outputFormat            string
+	search                  string
+	kfmClusterList          *kafkamgmtclient.EnterpriseClusterList
+	clustermgmtClusterList  []*clustersmgmtv1.Cluster
+	registeredClusters      []clusterRow
+	accessToken             string
+	clusterManagementApiUrl string
+
+	f *factory.Factory
+}
+
+func NewListClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   f.Localizer.MustLocalize("dedicated.list.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("dedicated.list.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("dedicated.list.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runListClusters(opts, f)
+		},
+	}
+
+	return cmd
+}
+
+func runListClusters(opts *options, f *factory.Factory) error {
+	err := listEnterpriseClusters(opts, f)
+	if err != nil {
+		return err
+	}
+	clist, err := getPaginatedClusterList(opts)
+	if err != nil {
+		return err
+	}
+	opts.clustermgmtClusterList = clist.Slice()
+	opts.registeredClusters = kfmListToClusterRowList(opts)
+	err = displayRegisteredClusters(opts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func listEnterpriseClusters(opts *options, f *factory.Factory) error {
+	conn, err := f.Connection()
+	if err != nil {
+		return err
+	}
+	ctx := context.Background()
+	api := conn.API()
+	cl := api.KafkaMgmtEnterprise().GetEnterpriseOsdClusters(ctx)
+	clist, response, err := cl.Execute()
+	if err != nil {
+		return err
+	}
+	opts.kfmClusterList = &clist
+	f.Logger.Info(response)
+	if len(opts.kfmClusterList.Items) == 0 {
+		return f.Localizer.MustLocalizeError("dedicated.listClusters.log.info.noClusters")
+	}
+	return nil
+}
+
+func createSearchString(opts *options) string {
+	searchString := ""
+	for idx, kfmcluster := range opts.kfmClusterList.Items {
+		if idx > 0 {
+			searchString += " or "
+		}
+		searchString += fmt.Sprintf("id = '%s'", kfmcluster.Id)
+	}
+	return searchString
+}
+
+func getPaginatedClusterList(opts *options) (*clustersmgmtv1.ClusterList, error) {
+	// get ids of clusters and create an ocm call filtering by those ids
+	ocmcl, err := clustermgmt.GetClusterListByIds(opts.f, opts.accessToken, opts.clusterManagementApiUrl, createSearchString(opts), len(opts.kfmClusterList.Items))
+	if err != nil {
+		return nil, err
+	}
+	return ocmcl, nil
+}
+
+func kfmListToClusterRowList(opts *options) []clusterRow {
+	var crl []clusterRow
+	for _, kfmcluster := range opts.kfmClusterList.Items {
+		for _, ocmCluster := range opts.clustermgmtClusterList {
+			if kfmcluster.Id == ocmCluster.ID() {
+				crl = append(crl, clusterRow{
+					Name:          ocmCluster.Name(),
+					ID:            kfmcluster.Id,
+					Status:        *kfmcluster.Status,
+					CloudProvider: ocmCluster.CloudProvider().ID(),
+					Region:        ocmCluster.Region().ID(),
+				})
+			}
+		}
+	}
```

Comment:
nitpick: consider building a map of `cluster_id ->  clustermgmtCluster` to improve search and get rid of the second loop .
Instead of line 114 you'll the have 
```go
ocmCluster := theMap[kfmcluster.Id]
crl = append(crl, clusterRow{
					Name:          ocmCluster.Name(),
					ID:            kfmcluster.Id,
					Status:        *kfmcluster.Status,
					CloudProvider: ocmCluster.CloudProvider().ID(),
					Region:        ocmCluster.Region().ID(),
				})
```